x/twap/api_test.go

```diff
@@ -148,6 +163,45 @@ func (s *TestSuite) TestGetArithmeticTwap() {
 			expTwap: ThreePlusOneThird.MulInt64(2),
 		},
 
+		"(3 record) start and end point to same record": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record, tPlus20sp2Record},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapInput(baseTime.Add(10*time.Second), baseTime.Add(10*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDec(5),
+		},
+		"(3 record) start and end exactly at record times, different records": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record, tPlus20sp2Record},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapInput(baseTime.Add(10*time.Second), baseTime.Add(20*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDec(5),
+		},
+		"(3 record) start at second record, end after third record": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record, tPlus20sp2Record},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapInput(baseTime.Add(10*time.Second), baseTime.Add(30*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDecWithPrec(35, 1), // 5 for 10s, 2 for 10s
+		},
+		"(3 record) start at second record, end after third record, sp1": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record, tPlus20sp2Record},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapInput(baseTime.Add(10*time.Second), baseTime.Add(30*time.Second), quoteAssetB),
+			expTwap:      sdk.NewDecWithPrec(35, 2), // 0.2 for 10s, 0.5 for 10s
+		},
+		// start in middle of first and second record, end in middle of second and third record
+		"(3 record) interpolate: in between second and third record": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record, tPlus20sp2Record},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapInput(baseTime.Add(15*time.Second), baseTime.Add(25*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDecWithPrec(35, 1), // 5 for 5s, 2 for 5 = 35 / 10 = 3.5
+		},
+		// interpolate in time closer to second record
+		"(3 record) interpolate: get twap closer to second record": {
+			recordsToSet: []types.TwapRecord{baseRecord, tPlus10sp5Record, tPlus20sp2Record},
+			ctxTime:      baseTime.Add(time.Minute),
+			input:        makeSimpleTwapInput(baseTime.Add(15*time.Second), baseTime.Add(30*time.Second), quoteAssetA),
+			expTwap:      sdk.NewDec(3), // 5 for 5s, 2 for 10s = 45 / 15 = 3
+		},
+
```

Comment:
Suggest adding a case where we interpolate over 3 records. I think most of the test cases land somewhere within 2nd and 3rd but do not include 1st to 3rd

Reply:
Good call! afaik @stackman27 is working on this case should soon be PRd

Reply:
Yup PR will be up very soon